### PR TITLE
Refactor range

### PR DESCRIPTION
**Pull request by patel-bhavin**
NO MERGE UNTILL this PR does : https://github.com/redcanaryco/atomic-red-team/pull/1457 

This PR makes sure that attack_range and attack_range_cloud have similar functionality and code base
 
These commands are fully functional and tested

- [x] python cloud_attack_range.py build
- [x] python cloud_attack_range.py destroy
- [x] python cloud_attack_range.py stop
- [x] python cloud_attack_range.py resume
- [x] python cloud_attack_range.py configure

**TO BE DEVELOPED/Tested in a NEW PR :**

StImulating AWS Attacks 

- [x] Added https://github.com/redcanaryco/atomic-red-team to submodules

- Clone the atomic-red-team repo, since AWS simulation require the atomic test file to be present on your local machine and set the `atomic_red_team_path` in the conf file, 

- The AWS Atomic tests have **not been merged** yet, you will need to clone this fork for testing this PR 

- [ ] AWS SIMULATION PR: https://github.com/redcanaryco/atomic-red-team/pull/1457

Simulate: only simulate a one AWS TTP at a time:

- [ ] python cloud_attack_range.py simulate -st T1136.003 

 Clean up simulation

- [ ] python cloud_attack_range.py simulate -st T1136.003 --clean_up yes

- dump command + wiki and README
- Test and develop Kubernetes  
- Introduce an Attack chain file feature

### CODE REVIEW

**Comment by P4T12ICK** [comment]
@patel-bhavin do we also want to add the change to Atomic Red Team into this PR?

**Comment by patel-bhavin** [comment]
> @patel-bhavin do we also want to add the change to Atomic Red Team into this PR?

i think we should wait on adding ART to this right away since that is also a WIP and i want feedback on that.

**Comment by P4T12ICK** [comment]
T

> > @patel-bhavin do we also want to add the change to Atomic Red Team into this PR?
> 
> i think we should wait on adding ART to this right away since that is also a WIP and i want feedback on that. but yes, that is the plan.

Then we need to disable the simulate command until ART simulation is ready. We shouldn't merge not working code into master.

**Comment by patel-bhavin** [comment]
@P4T12ICK : i disabled the simulate and other commands! When you get time, please test this PR!

File: cloud_attack_range.py [hunk 1]
```diff
@@ -82,54 +60,149 @@
...
+
+    actions_parser = parser.add_subparsers(title="Attack Range actions", dest="action")
+    configure_parser = actions_parser.add_parser("configure", help="configure a new attack range")
+    build_parser = actions_parser.add_parser("build", help="Builds attack range instances")
+    #simulate_parser = actions_parser.add_parser("simulate", help="Simulates attack techniques")
+    destroy_parser = actions_parser.add_parser("destroy", help="destroy attack range instances")
+    stop_parser = actions_parser.add_parser("stop", help="stops attack range instances")
+    resume_parser = actions_parser.add_parser("resume", help="resumes previously stopped attack range instances")
+    show_parser = actions_parser.add_parser("show", help="list machines")
+    # test_parser = actions_parser.add_parser("test")
```
Comment (d1vious) [inline cloud_attack_range.py:146]: I would remove these comment out sections, clutters up the code a bit.

File: cloud_attack_range.py [hunk 2]
```diff
@@ -82,54 +60,149 @@
...
+    # replay_parser.add_argument("-dn", "--dump_name", required=True,
+    #                            help="name for the dumped attack data")
+    # replay_parser.add_argument("--dump", required=False,
+    #                     help="name of the dump as defined in attack_data/dumps.yml")
+    # replay_parser.set_defaults(func=replay)
+
+    # # Test Arguments
+    # test_parser.add_argument("-tf", "--test_file", required=True,
+    #                          type=str, default="", help='test file for test command')
+    # test_parser.set_defaults(func=test)
```
Comment (d1vious) [inline cloud_attack_range.py:195]: this entire section seems like it can be yanked out.

**Comment by d1vious** [comment]
Noticed that the  README.md still references Leonidas as well as the diagram, can we update does before merging?

**Review by d1vious — CHANGES_REQUESTED** [review]
under `configure` this should be done automatically using boto:

```
--------------------------PLEASE NOTE-------------------------------

In order to have a fully functional Cloud Attack Range, you will need to set additional parameters in the cloud_attack_range.conf file

- Collect CloudTrail logs:
sqs_queue_url = <instructions in cloud_attack_range.conf.template>

- Simulate cloud atomics from Atomic Red Team:
atomic_red_team_path = <instructions in cloud_attack_range.conf.template>

--------------------------------------------------------------------------------

> configuration file was written to: /Users/jhernandez/Research/malware/splunk/attack_range_cloud/cloud_attack_range.conf, run `python cloud_attack_range.py build` to create a new cloud_attack_range
you can also edit this file to configure advance parameters
> setup has finished successfully ... exiting
```

also instructions on how to clone to bring down submodules solves the need to point to configs for atomic red team. Lets chat more tomorrow if you like on how to do this.

**Review by d1vious — APPROVED** [review]
:+1: amazing work amigo.

File: cloud_attack_range.py (hunk 2 above)
Comment (patel-bhavin) [reply cloud_attack_range.py:195]: will do in a new PR

File: cloud_attack_range.py (hunk 1 above)
Comment (patel-bhavin) [reply cloud_attack_range.py:146]: will do in a new PR

**Pull request merged**